unit-tests/log/test-multi-thread-logs.cpp

```diff
@@ -0,0 +1,221 @@
+    CAPTURE(ss.str());
+    CHECK_NOFAIL(max_time.count() < 10);
+    REQUIRE(max_time.count() < 50);
+    
+    // checking logs correctness - that all the integer's values have been logged
+    {
+        // preparing array of number of iterations * number of threads
+        std::array<char, number_of_iterations * 10> check_table;
+        // setting all cells to 0
+        memset(&check_table[0], 0, check_table.size());
+        // marking cells at the position of each log integer value to 1
+        std::ifstream check_file(log_file_path, std::ios::in);
+        if (check_file.good())
+        {
+            std::string line;
+            while (std::getline(check_file, line)) {
+                auto value_in_log = stoi(line.substr(line.find_last_of(" ")));
+                check_table.at(value_in_log - 1) = 1;
+            }
+        }
+        // checking that all cells have value 1, which means that all logs have been sent and received
+        std::for_each(check_table.cbegin(), check_table.cend(), [](const char c) {
+            REQUIRE(c == 1);
+            });
+        check_file.close();
+    }
+
+    // checking logs order in means of logging time
+    {
+        std::ifstream check_file(log_file_path, std::ios::in);
+        int overall_ms = 0;
+        int overall_logs = 0;
+        if (check_file.good())
+        {
+            // get data to calculate "ideal" line
+            // using average from 50 first logs as one point,
+            // and average of 50 last points as second point
+            auto first_50_pair = get_next_50_lines_time_val(check_file);
+            unsigned int first_50_avg_time_ms = first_50_pair.first;
+            int first_50_avg_value = first_50_pair.second;
+            
+            std::string line;
+            for (int i = 0; i < (number_of_iterations * 10) - 1 - 50 - 50; ++i)
+                getline(check_file, line);
+            
+            auto last_50_pair = get_next_50_lines_time_val(check_file);
+            unsigned int last_50_avg_time_ms = last_50_pair.first;
+            int last_50_avg_value = last_50_pair.second;
+
+            // clear and return to start of the file
+            check_file.clear();
+            check_file.seekg(0);
+
+            // go over each line and check that its logging time is not "far away" in means of time than the "ideal" time
+            int max_delta = 0;
+            while (getline(check_file, line)) {
+                auto value_in_log = stoi(line.substr(line.find_last_of(" ")));
+                auto log_time_ms = get_time_from_line(line);
+                auto log_ideal_time_ms = get_log_ideal_time(first_50_avg_time_ms, first_50_avg_value, last_50_avg_time_ms, last_50_avg_value, value_in_log);
+                auto delta = std::abs(log_ideal_time_ms - log_time_ms);
+                if (delta > max_delta) max_delta = delta;
+            }
+            CHECK_NOFAIL(max_delta < 15);
```

Comment:
This is critical - add a comment and specify threshold units

Reply:
would CAPTURE(max_delta) be enough for you?